src/gsy_e/models/state.py

```diff
@@ -668,11 +695,18 @@ def add_default_values_to_state_profiles(self, future_time_slots):
                                    ESSEnergyOrigin.LOCAL: 0.,
                                    ESSEnergyOrigin.EXTERNAL: 0.})
 
-    def market_cycle(self, past_time_slot, current_time_slot: DateTime, all_future_time_slots):
+    def market_cycle(self, past_time_slot, current_time_slot: DateTime,
+                     all_future_time_slots: List[DateTime]):
         """
         Simulate actual Energy flow by removing pledged storage and adding bought energy to the
         used_storage
         """
+        if GlobalConfig.FUTURE_MARKET_DURATION_HOURS:
+            # In case the future market is enabled, the future orders have to be deleted once
+            # the market becomes a spot market
+            del self.offered_buy_kWh[current_time_slot]
+            del self.offered_sell_kWh[current_time_slot]
+            all_future_time_slots.insert(0, current_time_slot)
```

Comment:
I think that this is not really required. If you backtrack `all_future_time_slots` you will see that it ends up in this method of the future markets:

```
    @property
    def market_time_slots(self) -> List[DateTime]:
        """Return list of all time slots of future markets."""
        return list(self.slot_bid_mapping.keys())
```
This returns a list with the dict keys, which is indeed the future markets, but the original object (the dict) is not modified, only the copy of the keys that are created. Also, it is not really correct to add the time slot of the spot market as the first timeslot of the future markets, therefore I think that you should not force this behavior here. 

Even without this line though, I think that the fix will work well.

Reply:
Hand to hand with the below proposal (assigning it to 0) seems to be working indeed, adapted!